### PR TITLE
Add tool selection

### LINKED ISSUE #20: Customisation of the dbCAN consensus
CAZomevolve provides a wrapper for dbCAN that retrieves the output from each protein function prediction tool (HMMER, DIAMOND, eCAMI and dbCAN-sub), as well as the consensus, where at least two tools agree upon the same CAZyme family annotation. CAZomevolve drops all CAZyme subfamily annotations from the incorporated tools as these refer to CAZyme clusters generated by dbCAN and do not directly translate to the subfamilies in CAZy.

However, the definition of the consensus (at least 2 tools) is immutable at the moment. Users may want to be able to define what the consensus (e.g. 1 or all 3 tools). This requires minor changes to the CAZomevolve command-line interface (CLI) as well as code within CAZomevolve.

## Changes to be made

- Update CLI - add new arg (`-N`, `--num_of_tools`) for defining the number of tools that need to agree upon a CAZyme family annotation for a given protein to be defined as a consensus annotation. Defaults to 2.
- Update `get_dbcan_consensus()` to take `args` as a parameter or a new parameter `num_of_tools` which is used to implement the new CLI args, and apply the args to allow customisation of the dbCAN consensus definition
- Update unit tests to include new parameter for `get_dbcan_consensus()` and test the new code in this function

### PR DESCRIPTION
**Pull request by PeterMBlack**
Adding the ability to select the number of dbCAN CAZyme annotation tools to find a consensus annotation

### CODE REVIEW

File: cazomevolve/utilities/parsers/get_dbcan_parser.py [hunk 1]
```diff
@@ -104,6 +104,14 @@ def build_parser(
         help="enable/disable deletion of exisiting files",
     )
 
+    parser.add_argument(
+        "--toolcount",
```
Comment (HobnobMancer) [inline cazomevolve/utilities/parsers/get_dbcan_parser.py]: Change arg name to `tool_count`
```suggestion
        "--tool_count",

```

**Review by HobnobMancer — CHANGES_REQUESTED** [review]
Change version number count in `cazomevolve.__init__.py`

File: cazomevolve/utilities/parsers/get_dbcan_parser.py (hunk 1 above)
Comment (PeterMBlack) [reply cazomevolve/utilities/parsers/get_dbcan_parser.py]: I have changed the argument to tool_count throughout the parser and python script, and changed the version number from 0.1.7.3 to 0.1.7.4 in the __init__.py

**Review by HobnobMancer — APPROVED** [review]
Looks good to me. Feel free to merge.

**Pull request merged**